org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalDescriptionProvider.java

```diff
@@ -70,11 +70,13 @@ public CompletionProposalDescriptionProvider(ICompilationUnit unit, CompletionCo
 		super();
 		fContext = context;
 		fUnit = unit;
+		fCollapsedTypes = new HashMap<>();
 	}
 
 	public CompletionProposalDescriptionProvider(CompletionContext context) {
 		super();
 		fContext = context;
+		fCollapsedTypes = new HashMap<>();
```

Comment:
if the maps are read-only, then `Collections.emptyMap()` would be better.

Reply:
I change the approach to add null check to avoid NPE, that's a more straightforward fix.